debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/processor/AbstractLogMinerEventProcessor.java

```diff
@@ -965,6 +970,38 @@ private class ParsedLobWriteSql {
      */
     protected abstract Scn getTransactionCacheMinimumScn();
 
+    /**
+     * Returns whether the transaction id has no sequence number component.
+     *
+     * Oracle transaction identifiers are a composite of:
+     * <ol>
+     *     <li>Undo segment number</li>
+     *     <li>Slot numbber of the transaction that generated the change</li>
+     *     <li>Sequence number of the transaction that generated the change</li>
+     * </ol>
+     *
+     * When Oracle LogMiner mines records, it is possible that when an undo operation is detected,
+     * often the product of a constraint violation, the LogMiner row will have the same explicit
+     * XID (transaction id) as the source operation that we should undo; however, if the record
+     * to be undone was mined in a prior iteration, Oracle LogMiner won't be able to make a link
+     * back to the full transaction's sequence number, therefore the XID value for the undo row
+     * will contain only the undo segment number and slot number, setting the sequence number to
+     * 4294967295 (aka -1 or 0xFFFFFFFF).
+     *
+     * This method explicitly checks if the provided transaction id has the no sequence sentinel
+     * value and if so, returns {@code true}; otherwise returns {@code false}.
+     *
+     * @param transactionId the transaction identifier to check, should not be {@code null}
+     * @return true if the transaction has no sequence reference, false if it does
+     */
+    protected boolean isTransactionIdWithNoSequence(String transactionId) {
+        return transactionId != null && transactionId.endsWith(NO_SEQUENCE_TRX_ID_SUFFIX);
```

Comment:
@Naros Is it possible that `transactionId` is `null`? If yes then I'd say the condition is not correct as in that case if will return `false` even if it is with no sequence.